src/browser/api/application.js

```
@@ -444,18 +451,44 @@ Application.revokeWindowAccess = function() {
 };
 
 Application.run = function(identity, configUrl = '') {
-
     if (!identity) {
         return;
     }
 
-    createAppObj(identity.uuid, null, configUrl);
+    const uuid = identity.uuid;
+    const app = createAppObj(uuid, null, configUrl);
+    const mainWindowOpts = _.clone(app._options);
+    let preloads = mainWindowOpts.preload;
+
+    // If preload option define, make sure it is either a string primitive or an array of objects with `url` props.
+    if (preloads) {
+        if (typeof preloads === 'string') {
+            preloads = [{ url: preloads }];
+        } else if (!Array.isArray(preloads) || preloads.find(preload => !isPreload(preload))) {
+            log.writeToLog(1, 'Expected `preload` option to contain a string primitive OR an array of objects with `url` props.', true);
```

will this silently fail?